sharding/data_modeling.rst

```diff
@@ -24,12 +24,90 @@ If your situation resembles either of these cases then the next step is to decid
 
 If you're migrating an existing database to the Citus multi-tenant architecture then some of your tables may lack a column for the application-specific tenant id. You will need to add one and fill it with the correct values. This will denormalize your tables slightly. For more details and a concrete example of backfilling the tenant id, see our guide to :ref:`Multi-Tenant Migration <transitioning_mt>`.
 
+Typical Multi-Tenant Scenario
+-----------------------------
+
+Most SaaS applications already have the notion of tenancy built into their data model. In the following, we will look at an example schema from the online advertising space. In this example, a web advertising platform has tenants that it refers to as accounts. Each account holds and tracks advertising clicks across various campaigns.
+
+.. code-block:: postgres
+
+  CREATE TABLE accounts (
+    id integer PRIMARY KEY,
+    name text NOT NULL,
+    image_url text NOT NULL
+  );
+
+  CREATE TABLE campaigns (
+    id integer NOT NULL,
+    account_id integer NOT NULL,
+    name text NOT NULL,
+    budget integer,
+    blacklisted_site_urls character varying[],
+
+    PRIMARY KEY (account_id, id),
+    FOREIGN KEY (account_id) REFERENCES accounts
+  );
+
+  CREATE TABLE ads (
+    id integer NOT NULL,
+    account_id integer NOT NULL,
+    campaign_id integer NOT NULL,
+    name text NOT NULL,
+    image_url text NOT NULL,
+    target_url text NOT NULL,
+    impressions_count bigint DEFAULT 0 NOT NULL,
+    clicks_count bigint DEFAULT 0 NOT NULL,
+
+    PRIMARY KEY (account_id, id),
+    FOREIGN KEY (account_id) REFERENCES accounts,
+    FOREIGN KEY (account_id, campaign_id) REFERENCES campaigns (account_id, id)
+  );
+
+  CREATE TABLE clicks (
+    id uuid DEFAULT gen_random_uuid() NOT NULL,
+    account_id integer NOT NULL,
+    ad_id integer NOT NULL,
+    clicked_at timestamp without time zone NOT NULL,
+    site_url text NOT NULL,
+    cost_per_click_usd numeric(20,10),
+    user_ip inet NOT NULL,
+    user_data jsonb NOT NULL,
+
+    PRIMARY KEY (account_id, id),
+    FOREIGN KEY (account_id) REFERENCES accounts,
+    FOREIGN KEY (account_id, ad_id) REFERENCES ads (account_id, id)
+  );
+
+  SELECT create_distributed_table('accounts',  'id');
+  SELECT create_distributed_table('campaigns', 'account_id');
+  SELECT create_distributed_table('ads',       'account_id');
+  SELECT create_distributed_table('clicks',    'account_id');
+
+JOINs and complex queries are supported within a tenant. For instance to count newly arriving clicks per campaign for account id=9700 we can join, group, and count.
```

Comment:
The other queries in example-ad-analytics involve yet more tables and I don't want to make the demo schema any longer. What I did instead was remove the `campaigns` table from the schema and add more explanation at the end. That brings up the ratio of payoff to preliminaries.